src/Microsoft.Extensions.Configuration.AzureKeyVault/IKeyVaultClient.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.Azure.KeyVault;
+
+namespace Microsoft.Extensions.Configuration.AzureKeyVault
+{
+    /// <summary>
+    /// Client class to perform cryptographic key operations and vault operations
+    /// against the Key Vault service.
+    /// Thread safety: This class is thread-safe.
+    /// </summary>
+    public interface IKeyVaultClient
```

Comment:
Is this interface primarily for testing?  I assume its not currently possible to Mock/derive from the existing AzureKeyVault for the tests?